### DISCUSSION TITLE
Yolov5 NMS calculation

### DISCUSSION THREAD

**Discussion by krishnakanagal** [ultralytics/yolov5#7029] [General]
I want to understand why the object confidence score is multiplied with the class confidence score in the non_max_suppression in yolov5?

`x[:, 5:] *= x[:, 4:5]  # conf = obj_conf * cls_conf`

Is there any benefit in doing this? Why is the class confidence score not used as is?

**Comment by glenn-jocher** [comment]
@krishnakanagal class confidence times object confidence is the total confidence that is used for NMS.

**Comment by nrupatunga** [comment]
@krishnakanagal It's the common practice and with the logic that if any one of the confidences is low, we make sure to deprioritize them during NMS and hence multiplication `obj_conf * cls_conf`

**Reply by TrueWodzu** [reply]
What do you mean by "deprioritize them", they are simply removed. This is counter-intuitive because if we are passing conf_thres as a parameter, then we are expecting to prune all boxes that are below that threshold, not some other threshold that is being calculated under the hood.
